### DISCUSSION TITLE
How does changing templates work?

### DISCUSSION THREAD

**Discussion by marcovanbeek** [poweradmin/poweradmin#524] [Q&A]
Hi,

I recently had a need to change some settings on a bunch of zones that were all based on the same template. In the end I did them all manually (16 zones) because I did not understand the implications of what I was about to do, and could not find out, so played safe. On that basis, I though if I asked the question here, maybe it would serve as the basis of some documentation.

Q: On the zone editing page there is a way of changing the template the zone is based on. If I change that to a new template, will the existing zone be overwritten by the information in the new template and therefore any changes to the original zone will be lost, or will the new information get added to the zone file? Secondly, if I change the template, will the zones automatically update in the background, or do I have to go into each zone and apply the new/revised template manually.

So these questions are as much "what is the correct answer" as much as "what happens at the moment". I could determine what happens now with some tests, but what I don't know is what SHOULD happen, so this is also a bit of a philosophical question. I will caveat this saying we are still on an older version of PowerDNS-Admin as our DNS servers are still on Ubuntu 20:04 so don't have PHP8 yet.

**Comment by edmondas** [comment]
When you change a zone's template:
  1. Existing records are overwritten - the system deletes template-based records and replaces them completely
  2. Manual records are preserved - only records originally from templates are replaced
  3. Changes are immediate - template application happens right when you change it on the edit page
  4. SOA records are updated with a new serial number

  Template sync behavior:
  - Manual sync only - zones do NOT automatically update when templates are modified
  - Sync tracking exists - the system tracks which zones need syncing after template changes
  - Bulk sync available - administrators can manually sync zones via "Update Zones" button

**Comment by edmondas** [comment]
I've updated the documentation to reflect the upcoming 4.0.0 version, which introduces new permissions for managing template creation and editing access. You can find more details at: https://docs.poweradmin.org/user-guide/dns-templates/
